pkg/genericapiserver/api/handlers/responsewriters/status.go

```diff
@@ -49,20 +48,14 @@ func apiStatus(err error) *metav1.Status {
 		//TODO: check for invalid responses
 		return &status
 	default:
-		status := http.StatusInternalServerError
-		switch {
-		//TODO: replace me with NewConflictErr
-		case storage.IsTestFailed(err):
```

Comment:
That's the big question. This code path looks like sneaked in. The official one is in `InterpretUpdateError`.